fintrack/src/context/UserContext.js

```diff
@@ -0,0 +1,59 @@
+import { createContext, useState, useContext } from 'react'
+import Cookies from 'js-cookie'
+
+import { AuthContext } from './AuthContext'
+const UserContext = createContext(null)
+
+// wrapper component that provides the context to its children
+export const UserProvider = (props) => {
+
+    // set authUser state from cookie sync
+    const cookie = Cookies.get('authenticatedUser')
+    const [authUser, setAuthUser] = useState(cookie ? JSON.parse(cookie) : null)
+
+
+    const { actions } = useContext(AuthContext)
+
+
+    // actions
+    const signIn = async (credentials) => {
+        const fetchOptions = {
+            method: 'POST',
+            headers: { 'Content-Type': 'application/json; charset=utf-8' },
+            body: JSON.stringify(credentials)
+        }
+
+
+        const response = await fetch('https://fintrackapp.onrender.com/api/v1/auth/login', fetchOptions)
+        if (response.status === 200) {
+            const { user, token } = await response.json()
+            actions.setToken(token)
+
+            setAuthUser(user)
+            Cookies.set('authenticatedUser', JSON.stringify(user), { expires: 1 })
+            return null
+        } else if (response.status === 401 || response.status === 400) {
+            const errorInfo = await response.json()
+            return errorInfo
+        } else if (response.status === 500) {
+            console.log('ErrorLogin', response.status)
+            const errorInfo = { status: 'Login Failed', msg: 'Something went wrong. Please try again later.'}
+            return errorInfo
+        }
+    }
```

Comment:
Again, very appreciative overall about the amount of response handling in your fetch calls! And also again, here's a `useFetch` opportunity!